pkg/synthetictests/disruption_test.go

```diff
@@ -0,0 +1,109 @@
+package synthetictests
+
+import (
+	"testing"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+)
+
+func Test_dnsOverlapDisruption(t *testing.T) {
+	events := []monitorapi.EventInterval{
+		{
+			Condition: monitorapi.Condition{
+				Locator: "disruption/openshift-api connection/new",
+				Message: "reason/DisruptionSamplerOutageBegan DNS lookup timeouts began",
+			},
+			From: time.Now(),
+			To:   time.Now().Add(1 * time.Minute),
+		},
+		{
+			Condition: monitorapi.Condition{
+				Locator: "disruption/openshift-api connection/new",
+				Message: "reason/DisruptionSamplerOutageBegan DNS lookup timeouts began",
+			},
+			From: time.Now().Add(2 * time.Minute),
+			To:   time.Now().Add(3 * time.Minute),
+		},
+		{
+			Condition: monitorapi.Condition{
+				Locator: "disruption/openshift-api connection/new",
+				Message: "reason/DisruptionSamplerOutageBegan DNS lookup timeouts began",
+			},
+			From: time.Now().Add(4 * time.Minute),
+			To:   time.Now().Add(5 * time.Minute),
+		},
+		{
+			Condition: monitorapi.Condition{
+				Locator: "disruption/openshift-api connection/new",
+				Message: "reason/DisruptionBegan disruption",
+			},
+			From: time.Now().Add(6 * time.Minute),
+			To:   time.Now().Add(7 * time.Minute),
+		},
+		{
+			Condition: monitorapi.Condition{
+				Locator: "disruption/openshift-api connection/new",
+				Message: "reason/DisruptionBegan disruption",
+			},
+			From: time.Now().Add(8 * time.Minute),
+			To:   time.Now().Add(9 * time.Minute),
+		},
+		{
+			Condition: monitorapi.Condition{
+				Locator: "disruption/openshift-api connection/new",
+				Message: "reason/DisruptionBegan disruption",
+			},
+			From: time.Now().Add(10 * time.Minute),
+			To:   time.Now().Add(11 * time.Minute),
+		},
+	}
+
+	testCases := []struct {
+		name      string
+		events    monitorapi.Intervals
+		expectErr bool
+	}{
+		{
+			name:      "No overlap between DNS and disruption",
+			events:    events,
+			expectErr: false,
+		},
+		{
+			name: "Overlap between DNS and disruption",
+			events: append(events, monitorapi.EventInterval{
+				Condition: monitorapi.Condition{
+					Message: "reason/DisruptionSamplerOutageBegan DNS lookup timeouts began",
+				},
+				From: time.Now().Add(5 * time.Minute),
+				To:   time.Now().Add(6 * time.Minute),
+			}),
+			expectErr: true,
+		},
+		{
+			name: "Overlap within 10 seconds between DNS and disruption",
+			events: append(events, monitorapi.EventInterval{
+				Condition: monitorapi.Condition{
+					Message: "reason/DisruptionSamplerOutageBegan DNS lookup timeouts began",
+				},
+				From: time.Now().Add(6*time.Minute + 5*time.Second),
+				To:   time.Now().Add(6*time.Minute + 15*time.Second),
+			}),
+			expectErr: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			testResults := testDNSOverlapDisruption(tc.events)
+
```

Comment:
Alternate approach here for consideration that is a little smaller

`
			// default no error
			expected := 1
			if tc.expectErr {
				expected = 2
			}
			assert.Equal(t, expected, len(testResults), "Test results did not match")`